File: Backend/src/models/plan-model.ts
import { Plans } from "@prisma/client"

export interface CreatePlanRequest {
    game_id: number
    plan_description: string
    plan_budget: number
    plan_currency: number 
    plan_tickets: number
}

export interface PlanResponse {
    plan_id: number
    game_id: number
    plan_description: string
    plan_budget: number
    plan_currency: number
    plan_tickets: number
}

export function toPlanResponse(plan: Plans): PlanResponse {
    return {
        plan_id: plan.plan_id,
        game_id: plan.game_id,
        plan_description: plan.plan_description,
        plan_budget: plan.plan_budget,
        plan_currency: plan.plan_currency,
        plan_tickets: plan.plan_tickets
    }
}

File: Frontend/app/src/main/java/com/visprog/starchive/models/PlanModel.kt
package com.visprog.starchive.models

import com.google.gson.annotations.SerializedName

data class GetAllPlanResponse(
    val data: List<PlanModel>
)

data class GetPlanResponse(
    val data: PlanModel
)

data class PlanModel(
    @SerializedName("plan_id") val planId: Int,
    @SerializedName("game_id") val gameId: Int,
    @SerializedName("plan_description") val planDescription: String,
    @SerializedName("plan_budget") val planBudget: Float,
    @SerializedName("plan_currency") val planCurrency: Float,
    @SerializedName("plan_tickets") val planTickets: Float
)

data class PlanRequest(
    @SerializedName("game_id") val gameId: Int,
    @SerializedName("plan_description") val planDescription: String,
    @SerializedName("plan_budget") val planBudget: Float,
    @SerializedName("plan_currency") val planCurrency: Float,
    @SerializedName("plan_tickets") val planTickets: Float
)